ls1intum/Artemis
# [Feature] Learning Goal Instructor View

### Checklist

### Motivation and Context


As an instructor it is useful to know how the class is doing on average on a learning goal. This PR implements this feature.

### Description


- New Query to calculate various statistics for exercises (average score etc)
- Used that query to calculate the average course progress for a learning goal 
- Changed client implementation of learning-goal-management to show the average course progress

### Steps for Testing


1. Set up a learning goal with connected exercise units (where results exist)
2. Check that learning goal management calculates the course average


### Test Coverage



Client
learning-goal-course-detail-modal.component.ts 100%

Server
LearningGoalService.java 98%
LearningGoalResource.java 90%
ExerciseService 91%



### Screenshots


![image](https://user-images.githubusercontent.com/29718932/102824844-e55f6600-43dd-11eb-8832-5bef7330b355.png)
![image](https://user-images.githubusercontent.com/29718932/102824863-f14b2800-43dd-11eb-829f-3495ede43778.png)
